pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go

```diff
@@ -2756,52 +2754,9 @@ func (lbaas *LbaasV2) ensureLoadBalancerDeleted(ctx context.Context, clusterName
 			}
 		}
 
-		// delete all members and pools
-		for _, poolID := range poolIDs {
-			// get members for current pool
-			membersList, err := openstackutil.GetMembersbyPool(lbaas.lb, poolID)
-			if err != nil && !cpoerrors.IsNotFound(err) {
-				return fmt.Errorf("error getting pool members %s: %v", poolID, err)
-			}
-			// delete all members for this pool
-			for _, member := range membersList {
-				mc := metrics.NewMetricContext("loadbalancer_member", "delete")
-				err := v2pools.DeleteMember(lbaas.lb, poolID, member.ID).ExtractErr()
-				if err != nil && !cpoerrors.IsNotFound(err) {
-					return mc.ObserveRequest(err)
-				}
-				mc.ObserveRequest(nil)
-				provisioningStatus, err := waitLoadbalancerActiveProvisioningStatus(lbaas.lb, loadbalancer.ID)
-				if err != nil {
-					return fmt.Errorf("timeout when waiting for loadbalancer to be ACTIVE after deleting member, current provisioning status %s", provisioningStatus)
-				}
-			}
-
-			// delete pool
-			mc := metrics.NewMetricContext("loadbalancer_pool", "delete")
-			err = v2pools.Delete(lbaas.lb, poolID).ExtractErr()
-			if err != nil && !cpoerrors.IsNotFound(err) {
-				return mc.ObserveRequest(err)
-			}
-			mc.ObserveRequest(nil)
-			provisioningStatus, err := waitLoadbalancerActiveProvisioningStatus(lbaas.lb, loadbalancer.ID)
-			if err != nil {
-				return fmt.Errorf("timeout when waiting for loadbalancer to be ACTIVE after deleting pool, current provisioning status %s", provisioningStatus)
-			}
-		}
-
 		// delete all listeners
-		for _, listener := range listenerList {
-			mc := metrics.NewMetricContext("loadbalancer_listener", "delete")
-			err := listeners.Delete(lbaas.lb, listener.ID).ExtractErr()
-			if err != nil && !cpoerrors.IsNotFound(err) {
-				return mc.ObserveRequest(err)
-			}
-			mc.ObserveRequest(nil)
-			provisioningStatus, err := waitLoadbalancerActiveProvisioningStatus(lbaas.lb, loadbalancer.ID)
-			if err != nil {
-				return fmt.Errorf("timeout when waiting for loadbalancer to be ACTIVE after deleting listener, current provisioning status %s", provisioningStatus)
-			}
+		if err := lbaas.deleteListeners(loadbalancer.ID, listenerList); err != nil {
```

Comment:
Have you tried deleting listener directly without deleting healthmonitor or pool?

Reply:
I've tried the following:
```
$ neutron lbaas-loadbalancer-delete kubeapi
listener 165725b2-dad7-4a3b-8c86-a153bfc2bde1 is using this loadbalancer

$ neutron lbaas-listener-delete 165725b2-dad7-4a3b-8c86-a153bfc2bde1
Deleted listener(s): 165725b2-dad7-4a3b-8c86-a153bfc2bde1

$ neutron lbaas-loadbalancer-delete kubeapi
pool a164402b-616f-47bd-a81a-a29732644642 is using this loadbalancer

$ neutron lbaas-pool-delete a164402b-616f-47bd-a81a-a29732644642
healthmonitor 00420307-e586-4662-b8a4-4b3ffe3637f6 is using this pool
```

So looks to me like you can delete the listener, but this won't delete the pools. And when you then try to delete the pools you have to delete the healthmonitors first

Reply:
Thanks